skfda/ml/regression/_fpls_regression.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Any, TypeVar, Union
+
+from sklearn.utils.validation import check_is_fitted
+
+from ..._utils._sklearn_adapter import BaseEstimator, RegressorMixin
+from ...misc.regularization import L2Regularization
+from ...preprocessing.dim_reduction import FPLS
+from ...representation import FDataGrid
+from ...representation.basis import Basis, FDataBasis
+from ...typing._numpy import NDArrayFloat
+
+InputType = TypeVar(
+    "InputType",
+    bound=Union[FDataGrid, FDataBasis, NDArrayFloat],
+)
+
+OutputType = TypeVar(
+    "OutputType",
+    bound=Union[FDataGrid, FDataBasis, NDArrayFloat],
+)
+
+
+class FPLSRegression(
+    BaseEstimator,
+    RegressorMixin[InputType, OutputType],
+):
+    r"""
+    Regression using Functional Partial Least Squares.
+
+    Parameters:
+        n_components: Number of components to keep. Defaults to 5.
+        regularization_X: Regularization for the calculation of the X weights.
+        weight_basis_X: Basis to use for the X block. Only
+            applicable if X is a FDataBasis. Otherwise it must be None.
+        weight_basis_Y: Basis to use for the Y block. Only
+            applicable if Y is a FDataBasis. Otherwise it must be None.
+        _integration_weights_X: One-dimensional array with the integration
+            weights for the X block.
+            Only applicable if X is a FDataGrid. Otherwise it must be None.
+        _integration_weights_Y: One-dimensional array with the integration
+            weights for the Y block.
+            Only applicable if Y is a FDataGrid. Otherwise it must be None.
+
+    Attributes:
+        coef\_: Coefficients of the linear model.
+        fpls\_: FPLS object used to fit the model.
+
+    Examples:
+        Fit a FPLS regression model with two components.
+
+        >>> from skfda.ml.regression import FPLSRegression
+        >>> from skfda.datasets import fetch_tecator
+        >>> from skfda.representation import FDataGrid
+        >>> from skfda.typing._numpy import NDArrayFloat
+        >>> X, y = fetch_tecator(return_X_y=True)
+        >>> fpls = FPLSRegression[FDataGrid, NDArrayFloat](n_components=2)
+        >>> fpls = fpls.fit(X, y)
+
+    """
+
+    def __init__(
+        self,
+        n_components: int = 5,
```

Comment:
Why this default?

Reply:
Seemed reasonable. I'll change it to 2, which is the default in `sklearn`, for consistency.

Reply:
I think the default should be the maximum, as in PCA.